### PR TITLE
[4] scheduler action log

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
Added action log entry  when a task is executed 

### Testing Instructions
enable log schedualer
run some tasks 

### Actual result BEFORE applying this Pull Request
N/A

### Expected result AFTER applying this Pull Request
image

### CODE REVIEW

**Comment by Kostelano** [comment]
Action outcome events must also be translated into other languages (Completed, Will Resume, Failure).

**Comment by alikon** [comment]
sure good spot

File: administrator/language/en-GB/plg_actionlog_joomla.ini [hunk 1]
```diff
@@ -60,3 +60,7 @@ PLG_ACTIONLOG_JOOMLA_EXTENSION_INSTALLED="User <a href='{accountlink}'>{username
 PLG_ACTIONLOG_JOOMLA_EXTENSION_UNINSTALLED="User <a href='{accountlink}'>{username}</a> uninstalled the {type} {extension_name}"
 PLG_ACTIONLOG_JOOMLA_EXTENSION_UPDATED="User <a href='{accountlink}'>{username}</a> updated the {type} {extension_name}"
 PLG_ACTIONLOG_JOOMLA_PLUGIN_INSTALLED="User <a href='{accountlink}'>{username}</a> installed the plugin <a href='index.php?option=com_plugins&task=plugin.edit&extension_id={id}'>{extension_name}</a>"
+; Scheduler
+PLG_ACTIONLOG_JOOMLA_TASK="Task <a href='{tasklink}'>{taskname}</a> #{times} runned for {duration} seconds <span class='badge bg-success'>{action}</span>"
```
Comment (Kostelano) [inline administrator/language/en-GB/plg_actionlog_joomla.ini]:
```suggestion
PLG_ACTIONLOG_JOOMLA_TASK_COMPLETED="Task <a href='{tasklink}'>{taskname}</a> #{times} runned for {duration} seconds <span class='badge bg-success'>{action}</span>"
```

File: plugins/actionlog/joomla/joomla.php [hunk 2]
```diff
@@ -1170,4 +1170,104 @@ public function onJoomlaAfterUpdate($oldVersion = null)
...
+		$context = 'Scheduler';
+		$user    = Factory::getUser();
+		$message = [
+			'action'   => 'Completed',
+			'taskname' => $event->get('title'),
+			'tasklink' => 'index.php?option=com_scheduler&view=tasks&filter[search]=id:' . $event->get('id'),
+			'duration' => round($snapshot['duration'], 2),
+			'times' => $event->get('times_executed'),
+		];
+		$this->addLog([$message], 'PLG_ACTIONLOG_JOOMLA_TASK', $context, $user->id);
```
Comment (Kostelano) [inline plugins/actionlog/joomla/joomla.php]:
```suggestion
		$this->addLog([$message], 'PLG_ACTIONLOG_JOOMLA_TASK_COMPLETED', $context, $user->id);
```

**Review by Kostelano — CHANGES_REQUESTED** [review]

**Comment by Quy** [comment]
Create a task.
Click `Run Test` button.

> Argument 1 passed to PlgActionlogJoomla::onTaskExecuteSuccess() must be an instance of Task, instance of Joomla\Component\Scheduler\Administrator\Task\Task given, called in \libraries\src\Plugin\CMSPlugin.php on line 285

File: administrator/language/en-GB/plg_actionlog_joomla.ini [hunk 3]
```diff
@@ -60,3 +60,10 @@ PLG_ACTIONLOG_JOOMLA_EXTENSION_INSTALLED="User <a href='{accountlink}'>{username
 PLG_ACTIONLOG_JOOMLA_EXTENSION_UNINSTALLED="User <a href='{accountlink}'>{username}</a> uninstalled the {type} {extension_name}"
 PLG_ACTIONLOG_JOOMLA_EXTENSION_UPDATED="User <a href='{accountlink}'>{username}</a> updated the {type} {extension_name}"
 PLG_ACTIONLOG_JOOMLA_PLUGIN_INSTALLED="User <a href='{accountlink}'>{username}</a> installed the plugin <a href='index.php?option=com_plugins&task=plugin.edit&extension_id={id}'>{extension_name}</a>"
+; Scheduler
+PLG_ACTIONLOG_JOOMLA_TASK_COMPLETED="Task <a href='{tasklink}'>{taskname}</a> #{times} runned for {duration} seconds <span class='badge bg-success'>{action}</span>"
+PLG_ACTIONLOG_JOOMLA_TASK_FAILURE="Task <a href='{tasklink}'>{taskname}</a> #{times} runned for {duration} seconds <span class='badge bg-danger'>{action}</span>"
+PLG_ACTIONLOG_JOOMLA_TASK_RESUME="Task <a href='{tasklink}'>{taskname}</a> #{times} runned for {duration} seconds <span class='badge bg-secondary'>{action}</span>"
+PLG_ACTIONLOG_JOOMLA_TASK_SUCCESS="Completed"
+PLG_ACTIONLOG_JOOMLA_TASK_KO="Failure"
```
Comment (Quy) [inline administrator/language/en-GB/plg_actionlog_joomla.ini]:
```suggestion
PLG_ACTIONLOG_JOOMLA_TASK_COMPLETED="Task <a href='{tasklink}'>{taskname}</a> #{times} executed for {duration} seconds <span class='badge bg-success'>{action}</span>"
PLG_ACTIONLOG_JOOMLA_TASK_FAIL="Failed"
PLG_ACTIONLOG_JOOMLA_TASK_FAILURE="Task <a href='{tasklink}'>{taskname}</a> #{times} executed for {duration} seconds <span class='badge bg-danger'>{action}</span>"
PLG_ACTIONLOG_JOOMLA_TASK_RESUME="Task <a href='{tasklink}'>{taskname}</a> #{times} executed for {duration} seconds <span class='badge bg-secondary'>{action}</span>"
PLG_ACTIONLOG_JOOMLA_TASK_SUCCESS="Completed"
```

**Comment by ditsuke** [comment]
This is something we've needed for a long time now, but something feels off about logging scheduler events to the "User Action Log". In my opinion these logs shouldn't come under user actions since the actions are being performed automatically by the Scheduler, instead they should go under a separate space for automated system actions unless the task is being executed manually.

**Pull request closed without merging**